src/main/java/org/apache/flink/benchmark/SerializationFrameworkBenchmarks.java

```diff
@@ -0,0 +1,451 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.benchmark;
+
+import org.apache.flink.api.common.ExecutionConfig;
+import org.apache.flink.api.common.typeinfo.TypeInformation;
+import org.apache.flink.api.common.typeinfo.Types;
+import org.apache.flink.api.java.tuple.Tuple2;
+import org.apache.flink.api.java.tuple.Tuple6;
+import org.apache.flink.api.java.typeutils.ResultTypeQueryable;
+import org.apache.flink.benchmark.functions.BaseSourceWithKeyRange;
+import org.apache.flink.streaming.api.environment.LocalStreamEnvironment;
+import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;
+import org.apache.flink.streaming.api.functions.sink.DiscardingSink;
+import org.apache.flink.types.Row;
+
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.OperationsPerInvocation;
+import org.openjdk.jmh.runner.Runner;
+import org.openjdk.jmh.runner.RunnerException;
+import org.openjdk.jmh.runner.options.Options;
+import org.openjdk.jmh.runner.options.OptionsBuilder;
+import org.openjdk.jmh.runner.options.VerboseMode;
+
+import java.util.Arrays;
+
+/**
+ * Benchmark for serializing POJOs and Tuples with different serialization frameworks.
+ */
+public class SerializationFrameworkBenchmarks extends BenchmarkBase {
+
+	private static final int RECORDS_PER_INVOCATION = 100_000;
```

Comment:
`100_000` records per invocation and judging from the results that you posted one invocation takes ~1 second, right? If you double the value, the results stay the same?

Reply:
true - they don't :(
with `300_000` I get the following results:
```
SerializationFrameworkBenchmarks.testAvroForcedSerializer      thrpt   30  129.923 ±  3.274  ops/ms
SerializationFrameworkBenchmarks.testAvroSchemaSerializer      thrpt   30  216.473 ± 34.747  ops/ms
SerializationFrameworkBenchmarks.testKryoSerializer            thrpt   30   70.413 ±  1.143  ops/ms
SerializationFrameworkBenchmarks.testKryoSerializerRegistered  thrpt   30  104.970 ±  1.541  ops/ms
SerializationFrameworkBenchmarks.testPojoSerializer            thrpt   30  344.357 ± 25.534  ops/ms
SerializationFrameworkBenchmarks.testPojoSerializerRegistered  thrpt   30  344.299 ± 22.487  ops/ms
SerializationFrameworkBenchmarks.testRowSerializer             thrpt   30  416.346 ± 29.662  ops/ms
SerializationFrameworkBenchmarks.testTupleSerializer           thrpt   30  440.312 ±  5.931  ops/ms
```
it's a strange thing that Kryo actually got slower (as the only benchmark!)

Reply:
looks like 300k is a good enough number: 100k and 200k have significantly lower numbers for Pojo, Row, and Tuple while the gain by 400k and 500k is not significant any more (at least on my PC)